### PR TITLE
variable: deprecated tidb_auto_analyze_partition_batch_size

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/53853

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by winoros — APPROVED** [review]

File: pkg/sessionctx/variable/tidb_vars.go [hunk 1]
```diff
@@ -1312,7 +1312,7 @@ const (
 	DefTiDBEnable1PC                               = false
 	DefTiDBGuaranteeLinearizability                = true
 	DefTiDBAnalyzeVersion                          = 2
-	DefTiDBAutoAnalyzePartitionBatchSize           = 128
+	DefTiDBAutoAnalyzePartitionBatchSize           = mysql.PartitionCountLimit
```
Comment (hi-rustin) [inline pkg/sessionctx/variable/tidb_vars.go:1315]: Please add the comment to mark it as depreciated.

File: pkg/sessionctx/variable/sysvar.go [hunk 2]
```diff
@@ -957,8 +957,11 @@ var defaultSysVars = []*SysVar{
 			}
 			AutoAnalyzePartitionBatchSize.Store(val)
 			return nil
-		}},
-
+		}, Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
+			vars.StmtCtx.AppendWarning(ErrWarnDeprecatedSyntaxNoReplacement.FastGenByArgs(TiDBAutoAnalyzePartitionBatchSize))
```
Comment (hi-rustin) [inline pkg/sessionctx/variable/sysvar.go:961]: Please add comments to mark TiDBAutoAnalyzePartitionBatchSize as deprecated.

File: pkg/sessionctx/variable/tidb_vars.go (hunk 1 above)
Comment (hawkingrei) [reply pkg/sessionctx/variable/tidb_vars.go:1327]: I have added comments.

File: pkg/sessionctx/variable/sysvar.go (hunk 2 above)
Comment (hawkingrei) [reply pkg/sessionctx/variable/sysvar.go:964]: I have added comments.

File: pkg/sessionctx/variable/sysvar.go [hunk 3]
```diff
@@ -959,8 +959,12 @@ var defaultSysVars = []*SysVar{
 			}
 			AutoAnalyzePartitionBatchSize.Store(val)
 			return nil
-		}},
-
+		}, Validation: func(vars *SessionVars, normalizedValue string, originalValue string, scope ScopeFlag) (string, error) {
+			// ```tidb_auto_analyze_partition_batch_size``` is deprecated and will be removed in the future.
```
Comment (hi-rustin) [inline pkg/sessionctx/variable/sysvar.go:963]:
```suggestion
```

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**